platform/entanglement/src/services/CopyTask.js

```diff
@@ -33,36 +33,48 @@ define(
          *
          * @param domainObject The object to copy
          * @param parent The new location of the cloned object tree
-         * @param persistenceService
          * @param $q
-         * @param now
          * @constructor
          */
-        function CopyTask (domainObject, parent, persistenceService, $q, now){
+        function CopyTask (domainObject, parent, policyService, $q){
             this.domainObject = domainObject;
             this.parent = parent;
+            this.firstClone = undefined;
             this.$q = $q;
             this.deferred = undefined;
-            this.persistenceService = persistenceService;
+            this.policyService = policyService;
             this.persisted = 0;
-            this.now = now;
             this.clones = [];
         }
 
         function composeChild(child, parent) {
             //Once copied, associate each cloned
             // composee with its parent clone
-            child.model.location = parent.id;
-            parent.model.composition = parent.model.composition || [];
-            return parent.model.composition.push(child.id);
+
+            parent.getModel().composition.push(child.getId());
+
+            //Check if the object being composed is a link
+            if (!child.getCapability("location").isLink()) {
```

Comment:
Hrm, the intention is to literally check whether the object is a link, it's not an indirect way of checking anything else, although you are correct that it will in effect distinguish between cloned and linked objects. Although most links get duplicated as fully-fledged objects (as per https://github.com/nasa/openmctweb/pull/249#discussion_r44058570 ), non-creatable types get composed as links, and we don't want to overwrite their location. If in the future we decided that links should be copied as links as well, this will still work in that instance. In any case, if an object is a link, we don't want to set its location.

I don't think it's particularly brittle, and it made sense to me to do the steps involved in composition in one place. Perhaps its a conceptual misunderstanding though, I tend to think of location being a step in the act of composition (of a new object anyway), but I suppose it's not really?